zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -104,7 +118,59 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final def catchAll[R1 <: R, E1, A1 <: A, B1 >: B](f: E => Http[R1, E1, A1, B1])(implicit
     @unused ev: CanFail[E],
   ): Http[R1, E1, A1, B1] =
-    self.foldHttp(f, Http.succeed, Http.empty)
+    self.foldHttp(f, Http.die, Http.succeed, Http.empty)
+
+  /**
+   * Recovers from all defects with provided function.
+   *
+   * '''WARNING''': There is no sensible way to recover from defects. This
+   * method should be used only at the boundary between `Http` and an external
+   * system, to transmit information on a defect for diagnostic or explanatory
+   * purposes.
+   */
+  @nowarn
+  final def catchAllDefect[R1 <: R, E1 >: E, A1 <: A, B1 >: B](
+    h: Throwable => Http[R1, E1, A1, B1],
+  ): Http[R1, E1, A1, B1] =
+    catchSomeDefect { case t => h(t) }
+
+  /**
+   * Recovers from all NonFatal Throwables.
+   */
+  final def catchNonFatalOrDie[R1 <: R, E2, A1 <: A, B1 >: B](
+    h: E => Http[R1, E2, A1, B1],
+  )(implicit ev1: CanFail[E], ev2: E <:< Throwable): Http[R1, E2, A1, B1] =
```

Comment:
This can be implemented using `catchSome` also. I think the code will look simpler.

Reply:
Done.